src/shogun/mathematics/RandomMixin.h

```diff
@@ -0,0 +1,74 @@
+#ifndef __RANDOMMIXIN_H__
+#define __RANDOMMIXIN_H__
+
+#include <shogun/lib/config.h>
+
+#include <random>
+
+namespace shogun
+{
+	static inline void set_seed_callback(CSGObject*, int32_t);
+
+	template <
+	    typename Parent, typename PRNG = std::mt19937_64,
+	    typename RandomDevice = std::random_device>
+	class RandomMixin : public Parent
+	{
+	public:
+		template <typename... T>
+		RandomMixin(T... args) : Parent(args...)
+		{
+			init();
+		}
+
+		void set_random_seed()
+		{
+			RandomDevice random_device;
+			set_seed(random_device());
+		}
+
+		void set_seed(const int32_t seed)
+		{
+			m_seed = seed;
+			reinit();
+			set_seed_callback(this, seed);
+		}
+
+		void reinit()
```

Comment:
use-case?

Reply:
when seed is set, we create a new PRNG object with the new seed, since the PRNG concept doesn't provide any other way.

Reply:
what i meant is: wouldn't having `set_seed` be enough? that reinits the PRNG. do we need a separate reinit (that is for the same seed)?

Reply:
yes we need it. in the callback function we tell the object to `reinit` once seed is changed via `put`. there is a bug here, I forgot to register `reinit`

Reply:
I got rid of `reinit` and `put_quietly`